pkg/route/ingress/ingress.go

```diff
@@ -435,15 +439,45 @@ func (c *Controller) sync(key queueKey) error {
 			return err
 		}
 		data = []byte(fmt.Sprintf(`[{"op":"replace","path":"/spec","value":%s}]`, data))
-		_, err = c.client.Routes(route.Namespace).Patch(context.TODO(), route.Name, types.JSONPatchType, data, metav1.PatchOptions{})
+		_, err = c.routeClient.Routes(route.Namespace).Patch(context.TODO(), route.Name, types.JSONPatchType, data, metav1.PatchOptions{})
 		if err != nil {
 			errs = append(errs, err)
 		}
 	}
 
 	// purge any previously managed routes
 	for _, route := range old {
-		if err := c.client.Routes(route.Namespace).Delete(context.TODO(), route.Name, metav1.DeleteOptions{}); err != nil && !kerrors.IsNotFound(err) {
+		if err := c.routeClient.Routes(route.Namespace).Delete(context.TODO(), route.Name, metav1.DeleteOptions{}); err != nil && !kerrors.IsNotFound(err) {
+			errs = append(errs, err)
+		}
+	}
+
+	// reflect route status to ingress status
```

Comment:
If I'm following these changes correctly, route status should only be reflected back to the parent ingress object if `len(matches) > 0`, right? If that's true, would it be worth encasing lines 456->483 in an if statement that checks for the length of `matches`?

Reply:
I don't think that's right.  What if the ingress is created in such a way that corresponding routes are created and their statuses reflected back to the ingress, and then the ingress is modified in such a way that the routes are removed?  The status should be cleared in that case.  Hm, maybe this needs a test case...

Reply:
Ah, I see,I missed that edge case.

In terms of adding a test case, perhaps one of the `delete route when` test cases could be enhanced to verify that Ingress status conditions are updated properly when corresponding routes are deleted? The current code coverage for the changes in this PR is fantastic, but I think this potential flow should be visible from the test cases (or at least some in-line comments).

Reply:
Done.  Thanks!